src/backend/access/index/genam.c

```diff
@@ -616,11 +616,11 @@ systable_endscan(SysScanDesc sysscan)
 		UnregisterSnapshot(sysscan->snapshot);
 
 	/*
-	 * Reset the bsysscan flag at the end of the systable scan.  See detailed
+	 * Decrement the sysscan_depth counter at the end of the systable scan.  See detailed
 	 * comments in xact.c where these variables are declared.
 	 */
 	if (TransactionIdIsValid(CheckXidAlive))
-		bsysscan = false;
```

Comment:
Non-actionable note: One problem that we run into with citus with such depth counters is that when an error happens they are not decremented correctly. I'm not sure if that applies here. It seems like ResetLogicalStreamingState probably takes care of it.

The fix we usually do in citus is to use PG_TRY + PG_FINALLY, but that's not possible here as the error could happen anywhere inbetween the systable_beginscan and the systable_endscan.